vm/src/stdlib/signal.rs

```diff
@@ -0,0 +1,183 @@
+use crate::obj::objint::PyIntRef;
+use crate::pyobject::{IdProtocol, PyObjectRef, PyResult};
+use crate::vm::VirtualMachine;
+
+use std::sync::atomic::{AtomicBool, Ordering};
+
+use num_traits::cast::ToPrimitive;
+
+use arr_macro::arr;
+
+#[cfg(unix)]
+use nix::unistd::alarm as sig_alarm;
+
+use libc;
+
+#[cfg(not(windows))]
+use libc::{SIG_DFL, SIG_ERR, SIG_IGN};
+
+#[cfg(windows)]
+const SIG_DFL: libc::sighandler_t = 0;
+#[cfg(windows)]
+const SIG_IGN: libc::sighandler_t = 1;
+#[cfg(windows)]
+const SIG_ERR: libc::sighandler_t = !0;
+
+const NSIG: usize = 64;
+
+// We cannot use the NSIG const in the arr macro. This will fail compilation if NSIG is different.
+static mut TRIGGERS: [AtomicBool; NSIG] = arr![AtomicBool::new(false); 64];
+
+extern "C" fn run_signal(signum: i32) {
+    unsafe {
+        TRIGGERS[signum as usize].store(true, Ordering::Relaxed);
+    }
+}
+
+#[derive(Debug)]
+enum SigMode {
+    Ign,
```

Comment:
Does this abbreviate `Ignore`? We might want to use `Ignore`, `Default` and `Handler` here.

Reply:
I removed the enum as it is not needed now that I use the same code for windows and linux.